tests/monotouch-test/ARKit/ARPointCloudTest.cs

```diff
@@ -0,0 +1,97 @@
+﻿//
+// Unit tests for ARPointCloud
+//
+// Authors:
+//	Vincent Dondain <vidondai@microsoft.com>
+//
+// Copyright 2017 Microsoft. All rights reserved.
+//
+
+#if XAMCORE_2_0 && !__TVOS__ && !__WATCHOS__ && !MONOMAC
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using ARKit;
+using AVFoundation;
+using Foundation;
+using NUnit.Framework;
+using ObjCRuntime;
+using OpenTK;
+
+namespace MonoTouchFixtures.ARKit {
+
+	class ARPointCloudPoker : ARPointCloud {
+
+		GCHandle vectorArrayHandle;
+		GCHandle identifiersHandle;
+		Vector3 [] vectorArray;
+		ulong [] identifiers;
+
+		public ARPointCloudPoker () : base (IntPtr.Zero)
+		{
+		}
+
+		public override nuint Count {
+			get {
+				return 2;
+			}
+		}
+
+		protected unsafe override IntPtr GetRawPoints ()
+		{
+			vectorArray = new Vector3 [] { new Vector3 (1, 2, 3), new Vector3 (4, 5, 6) };
+			vectorArrayHandle = GCHandle.Alloc (vectorArray, GCHandleType.Pinned);
```

Comment:
This leaks when GetRawPoints is called again, because it overwrites the previous GCHandle.